tag-release.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+hub_installed=$(command -v hub)
+if [[ ! $hub_installed ]]; then
+	echo "Please install the hub command line tool before running this script."
+	echo "https://github.com/github/hub"
+	exit 1
+fi
+
+current_branch=$(git branch --show-current)
+if [[ $current_branch != "master" ]]; then
+	echo "This script should only be run from the master branch. Aborting."
+	exit 1
+fi
+
+git_changes=$(git status -s | wc -l)
+if [[ $git_changes != 0 ]]; then
+	echo "The master branch has been modified."
+	echo "Please revert the changes or move them to another branch before running this script."
+	exit 1
+fi
+
+git fetch
+ahead=$(git status -sb | grep ahead -c)
+if [[ $ahead != 0 ]]; then
+	echo "Your local master branch is ahead of the remote master branch. Aborting."
+	exit 1
+fi
+
+git pull origin master
+
+new_version=v$(cat Cargo.toml | grep -Po -m 1 '\d+\.\d+\.\d+')
+
+git tag -a $new_version HEAD -m "Release $new_verion"
```

Comment:
My preference would be to create a signed git tag so that we can add a bit of trust that the release was produced from a trusted source. 

Also this script assumes that you want to tag `HEAD` which may not be the case if new commits were pulled/fetched or tagging a patch. I would recommend we assume the current status of the git repo to be correct and not try and change it.

```
git tag -a -s $new_version -m "Release $new_version"
```